R/input_to_reference_feature.R

```diff
@@ -1,81 +1,268 @@
+# A function to find the most terminal feature given ids or coordinates
 #' @title Find a Refernece Feature
-#' @param network table from network file default is NULL
-#' @param id hydrofabric id (relevant only to nextgen fabrics)
-#' @param comid NHDPlusV2 COMID
-#' @param hl_id hydrolocation URI (relevant only to nextgen fabrics)
-#' @param poi_id POI identifier
-#' @param nldi_feature list with names 'featureSource' and 'featureID' where 'featureSource' is derived from the "source" column of the response of dataRetrieval::get_nldi_sources() and the 'featureID' is a known identifier from the specified 'featureSource'.
-#' @param xy Location given as vector of XY in CRS 4326 (longitude, latitude)
+#' @param network table from network file default is NULL. datatype: dataframe e.g., CONUS_Net file
+#' @param id hydrofabric id (relevant only to nextgen fabrics). datatype: string / vector of strings e.g., 'wb-10026' or c('wb-10026', 'wb-10355') 
+#' @param comid NHDPlusV2 COMID. datatype: int / vector of int e.g., 61297116 or c(61297116 , 6129261) 
+#' @param hl_id hydrolocation id (relevant only to nextgen fabrics). datatype: int / vector of int e.g., 01236 or c(01236 , 01244) 
+#' @param hl_uri hydrolocation URI (relevant only to nextgen fabrics). datatype: string / vector of string / a url e.g., HUC12-010100100101 or c(HUC12-010100100101 , HUC12-010100110104) 
+#' @param poi_id POI identifier. datatype: int / vector of int e.g., 266387 or c(266387, 266745)
+#' @param nldi_feature list with names 'featureSource' and 'featureID' where 'featureSource' is derived from the "source" column of the response of dataRetrieval::get_nldi_sources() and the 'featureID' is a known identifier from the specified 'featureSource'. datatype: a url e.g., 'https://labs.waterdata.usgs.gov/api/nldi/linked-data/census2020-nhdpv2'
+#' @param xy Location given as vector of XY and CRS (e.g., 4326) (longitude, latitude, crs)
 
+# Inputs 
 input_to_reference_feature = function(net, 
                                       id, 
                                       comid, 
-                                      hl_id, 
+                                      hl_id,
+                                      hl_uri, 
                                       poi_id,
                                       nldi_feature, 
-                                      xy){
+                                      xy) {
   
+  # Initialize as Null variables an default varaibles
   toid <- divide_id <- hf_hydroseq <- hf_id <- hydroseq <- vpu <- NULL
-  
+  s3 = 's3://lynker-spatial/hydrofabric'
```

Comment:
These three should be exposed as function parameters rather then hardcoded internally

Reply:
Fixed